.github/workflows/test-and-build.yaml

```diff
@@ -11,7 +11,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        go: ["1.16"]
+        go: ["1.17"]
```

Comment:
any specific reason for this change? as 1.17 is the latest release 3monts back it's good to stick to 1.16 for a while.

Reply:
Considering we are not seeing any issue with thi verison of GO, isnt it good to be on latest version?

Reply:
as there is know the hard requirement for it and it's a recent release. I would like to stick to 1.16 for a while

Reply:
reverted